client/apollo/js/View/Track/AnnotTrack.js

```diff
@@ -2294,6 +2331,71 @@ define([
                     'class': "annotation_info_editor_button"
                 }, commentButtons);
 
+                var replacementDiv = dojo.create("div", {'class': "annotation_info_editor_section"}, content);
+                var replacementLabel = dojo.create("div", {
+                    'class': "annotation_info_editor_section_header",
+                    innerHTML: "Replace model(s)"
+                }, replacementDiv);
+
+             
+                var replacementsTable = dojo.create("div", {
+                    'class': "replacement",
+                    id: "replacement_" + (selector ? "child" : "parent")
+                }, replacementDiv);
+                var replacementButtonsContainer = dojo.create("div", {style: "text-align: center;"}, replacementDiv);
+                var replacementButtons = dojo.create("div", {'class': "annotation_info_editor_button_group"}, replacementButtonsContainer);
+                var addreplacementButton = dojo.create("button", {
+                    innerHTML: "Add",
+                    'class': "annotation_info_editor_button"
+                }, replacementButtons);
+                var deletereplacementButton = dojo.create("button", {
+                    innerHTML: "Delete",
+                    'class': "annotation_info_editor_button"
+                }, replacementButtons);
+
+
+                var trackNames = Object.keys(JBrowse.trackConfigsByName);
+                var replacements = new Select({
+                    name: "replacementSelection",
+                    options: array.map(trackNames, function(trackName) {
+                        return {value: track.browser.trackConfigsByName[trackName].label, label: track.browser.trackConfigsByName[trackName].key}
+                    })
+                });
+
+
+                replacements.placeAt(replacementDiv).startup();
+                track.featureReplace = track.featureReplace ||new Select({
+                    name: "featureSelection",
+                    options: []
+                });
+                on(replacements, "change", function(selection) {
+                    console.log(track.browser.trackConfigsByName,selection,replacements);
+                    track.browser.getStore(track.browser.trackConfigsByName[selection].store, function(store) {
+                        console.log("Received store",store);
+                        store.getFeatures({ref: feature.afeature.sequence, start: feature.get('start'), end: feature.get('end')}, function(f) {
```

Comment:
I keep getting an error here that ```feature``` is undefined.  Sometimes this seems to work and sometimes it doesn't.  What "seems" to be happening is that it is requesting the unique_id for the gene instead of the transcript and so when it requests the feature up above it might be requesting for the wrong one, since I don't see a specific 'parent' vs 'child' id for these elements.  My guess is that if you get rid of the "gene" replace model, which you need to do anyway, this might go away.